portfolio/src/main/java/com/google/sps/validators/TextValidator.java

```diff
@@ -0,0 +1,25 @@
+package com.google.sps.validators;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class TextValidator {
+  // Allow only alphanumeric chars and few extra symbols.
+  private static final String validPattern = "^([A-z]|[0-9]|[\\?!,.(\" ])*$";
+
+  private final int lengthLimit;
+
+  private final String text;
+
+  public TextValidator(String text, int lengthLimit) {
+    this.text = text;
+    this.lengthLimit = lengthLimit;
+  }
+
+  public boolean validate() {
+    Pattern pattern = Pattern.compile(validPattern);
```

Comment:
Side-note: Optimization here for performance is not the goal. I'm not sure whether Java caches compiled regular expressions. But assuming that Java does not cache compiled regular expressions, and assuming that we cared here about performance, then I would move the compiled pattern into a class variable (compile once in constructor), move the text to be validated into method parameter (so you never need to construct new TextValidator objects). As I said, performance isn't critical here, so I'm just sharing thoughts.

Reply:
Thank you sharing this one!